tests/__parser__/test_translator.py

```diff
@@ -0,0 +1,126 @@
+        )
+    return assert_flag
+
+
+def test_translator_basic():
+    target_str = "required"
+    translator = Translator(target_str)
+    result_arr = translator.translate()
+    assert instanceChecker(target_str, result_arr, False)
+
+    target_str = "required|min:30"
+    translator = Translator(target_str)
+    result_arr = translator.translate()
+    assert instanceChecker(target_str, result_arr, False)
+
+    target_str = "required|min:30|max:40|mail"
+    translator = Translator(target_str)
+    result_arr = translator.translate()
+    assert instanceChecker(target_str, result_arr, False)
+
+    target_str = "mail|between:10,20"
+    translator = Translator(target_str)
+    result_arr = translator.translate()
+    assert instanceChecker(target_str, result_arr, False)
+
+
+def test_translator_mid():
+    target_arr = [
+        ["required"],
+        ["required", "mail", "max:30"],
+        ["required", "min:20", "max:30"],
+    ]
+    target_arr.append(["required", "between:10,20"])
+    for _ in target_arr:
+        target_str = _
+        translator = Translator(target_str)
+        result_arr = translator.translate()
+        assert instanceChecker(target_str, result_arr, True)
+
+
+def test_translator_single():
+    min_value, max_value = 10, 20
+    target_str = "between:" + str(min_value) + "," + str(max_value)
+    translator = Translator(target_str)
+    result_arr = translator.translate()
+    assert (result_arr[0].max_value == max_value) and (
+        result_arr[0].min_value == min_value
+    )
+
+    # min
+    target_str = "min:" + str(min_value)
+    translator = Translator(target_str)
+    result_arr = translator.translate()
+    assert result_arr[0].min_value == min_value
+
+    # max
+    target_str = "max:" + str(max_value)
+    translator = Translator(target_str)
+    result_arr = translator.translate()
+    assert result_arr[0].max_value == max_value
+
+
+# many features checking value
+def test_translator_many():
+    min_value, max_value = 10, 20
+    # between min
+    target_str = "between:" + str(min_value) + "," + str(max_value)
+    target_str += "|min:" + str(min_value)
+    translator = Translator(target_str)
+    result_arr = translator.translate()
+
+    assert (
+        (result_arr[0].max_value == max_value)
+        and (result_arr[0].min_value == min_value)
+        and ((result_arr[1].min_value == min_value))
+    )
+
+    # betweeen max
+    target_str = "between:" + str(min_value) + "," + str(max_value)
+    target_str += "|max:" + str(max_value)
+    translator = Translator(target_str)
+    result_arr = translator.translate()
+
+    assert (
+        (result_arr[0].max_value == max_value)
+        and (result_arr[0].min_value == min_value)
+        and ((result_arr[1].max_value == max_value))
+    )
+
+    # betweeen max min
+    target_str = "between:" + str(min_value) + "," + str(max_value)
+
+    target_str += "|min:" + str(min_value) + "|max:" + str(max_value)
```

Comment:
why not use python's f-strings? I think it looks better
Old way: 
```python 
target_str = "between:" + str(min_value) + "," + str(max_value)
target_str += "|min:" + str(min_value) + "|max:" + str(max_value)
```
f-strings: 
```python
target_str = f"between:{min_value},{max_value}|min:{min_value}|max:{max_value}"
```

Reply:
Thanks man! Will change for sure. I do not see problem here but i will change. Next time y can change it for yourself :D